packages/new-deploy/scripts/deploy.ts

```diff
@@ -0,0 +1,34 @@
+import DeployContext from '../src/DeployContext';
+import * as config from '@nomad-xyz/configuration';
+import * as ethers from 'ethers';
+import { NonceManager } from '@ethersproject/experimental';
+import fs from "fs";
+
+run();
+
+/*
+* get published config
+* modify to add a network
+* run the deploy script
+* re-publish config
+* */
+const config = {}; // TODO: get published (or local??) config
+const DEPLOYER_PRIVATE_KEY = ""; // TODO: get from .env - also, do we need different private keys for each chain??
+const OVERRIDES = {}; // TODO write local overrides config
+
+async function run() {
+    const deployContext = new DeployContext(config);
+    for (const network of deployContext.networks) {
+        const rpc = deployContext.data.rpcs[network][0]; // TODO: note: we can't publish private RPCs
+        const provider = new ethers.providers.JsonRpcProvider(rpc);
```

Comment:
there's a shortcut for this:
`context.registerRpcProvider(rpc)`

Reply:
guess it doesn't make as much sense if you have to manually instantiate the wallet anyway 🤔

Reply:
hmm but you're right it should actually be calling `context.registerRpcProvider(rpc)` when it constructs the `DeployContext`. i'll do `deployContext.mustGetProvider` instead of instantiating it separately.

Reply:
https://github.com/nomad-xyz/monorepo/pull/91/commits/69583fc3a930a74fbfe5d758a9c58f6d8d1f92be